### ISSUE TITLE
[MU4 Issue] Changing default time stretch of fermata in MU4 to 200% is a regression on MU3

### ISSUE THREAD

**Issue by SteveBlower** [musescore/MuseScore#15569]
**Describe the bug**

The default time stretch in MU 3 was 100%.  In MU4 it was changed to 200%.  Having a default time stretch of anything other than 100% is a great inconvenience when there are fermata on multiple instruments as Musescore honours the longest of them.  If a duration shorter than default 200% is required then the time stretch on ALL of the simultaneous fermata have to be shortened for the shortening to take effect in playback.

**Additional context**
Default was changed as a "fix" for this closed issue #13464

**Comment by cbjeukendrup** [comment]
Related to #15329.

Basically, the problem is that we need to determine which fermatas on different staves / voices belong together, so we need to "consolidate" those. Getting this _perfectly_ right in every possible case seems non-trivial to me, but everything is solvable. We may also want to provide a way to control this to the user. 

Keeping the default time stretch at 100% would be annoying because then fermatas won't seem to work at all (until you figure out that you need to enable a mystery thing called "time stretch" in some hidden popup somewhere, as a new user might experience it). 
And unless there is any other use case not described here yet, the only use case would be a workaround for the "consolidating" problem. I'm not a fan of making sub-optimal changes in favour of a workaround for another bug; I'd rather focus on fixing the other bug.

By the way, for inspiration: Dorico's solution to the consolidating problem is a very rigid one: they make fermatas a kind of system object, that is mandatorily applied to all voices on all staves. You can then move then to a different note for each voice individually, but hiding is not possible. Not ideal either, but still quite a nice solution.

**Comment by oktophonie** [comment]
Dorico's system, dare I say it, is very sensible; fermatas _should_ be a system object (and appear on all staves), though we do need to be able to override this behaviour.

**Comment by MarcSabatella** [comment]
A complication with any sort of system behavior is, the fermatas may well not be on the same beat.  Like, one staff might have a whole note with fermata, another four quarters with a fermata on the last.  So the first fermata is actually on beat one, the last on beat four.  This inconvenient fact causes any number of complications in the design.

If we didn't want to go the full-on Dorico approach, another possibility is to simply make any change to one fermata apply to all fermatas within the beat range affected.  Kind of like how key signatures or barlines are really per-staff internally, but most operations on them nevertheless are propagated to all staves.

**Comment by cbjeukendrup** [comment]
> A complication with any sort of system behavior is, the fermatas may well not be on the same beat.

This problem is solved by Dorico's model to some extent, but actually in a slightly different way than I remembered. Fermatas do have a "tick" (to use MuseScore's terminology) that is the same for all staves, and on which note the fermata appears depends only on this tick:

https://user-images.githubusercontent.com/48658420/209572406-7c4f7320-53a8-4951-9222-aaa9212bf274.mov

That makes it impossible to create a situation like this:
<img width="220" alt="Scherm­afbeelding 2022-12-26 om 20 00 14" src="https://user-images.githubusercontent.com/48658420/209576858-82c0e762-cf38-4467-88a3-89a021a4b50f.png">
while I think that this is not a nonsensical situation.

So maybe we need to choose a slightly different solution.

**Comment by SteveBlower** [comment]
> A complication with any sort of system behavior is, the fermatas may well not be on the same beat. Like, one staff might have a whole note with fermata, another four quarters with a fermata on the last.

However, all fermatas should end at the same time.  Think what a conductor would have to indicate in both those cases - beat 1, 2, 3, hold on 4 ... and release for beat 1 of the next measure.

The "impossible" example in @cbjeukendrup's post _is_  nonsensical, I would suggest.  What could the conductor do - instrument 1 has a fermata on beat 1 so hold beat 1, meanwhile instruments 2 and 3 are looking puzzled - when is beat 2 going to happen?  Or don't hold beat 1, but hold beat 2, meanwhile instrument 1 is now looking puzzled - what happened to the fermata?  Or hold both beats 1 and 2 and everyone looks puzzled!

It looks like Dorico's method also allows nonsensical notation to be produced with the example given requiring a hold somewhere or perhaps on 2 different beats but no way for the  conductor or players to know just by reading the score.

The addition of a fermata to note or rest X should have no effect on the duration of notes that end before note or rest X.  A note or rest that also ends at the same time as note X must also have a fermata.  A note that would otherwise end after note X must be split into shorter durations so that a there is a note that ends at the same time as note X but which is tied to other notes to complete the desired duration. 

Perhaps this issue should migrate to the discussion area so that the precise logic can be explored further.

**Comment by wizofaus** [comment]
I agree @cbjeukendrup's second example (where two players have the same duration note on the same beat, one with a fermata and one without) is something you couldn't even expect human players to make sense of, so to me it doesn't really matter what the computer playback is.   In general the rule seems pretty simple to me - extra time should always be added to the *end* of any note with a fermata, and the amount of extra time depends on the shortest note at that beat position.

**Comment by SteveBlower** [comment]
@wizofaus Can you clarify what you mean by  

> the amount of extra time depends on the shortest note at that beat position

Do you mean that the duration should be calculated as a %age of the duration of the shortest note **ending** at that beat position? - i.e. if you have a half note on beats 1 and 2 and a quarter note on beat 2, both of which have a fermata, the duration should only take into account the properties of the fermata attached to the quarter note and if say, the time stretch of the fermata on the half note is set at 250% and the time stretch of the fermata on the quarter note is set at 150%, the duration would be extended by the duration of an eighth note.   Would this need someway of indicating to users which fermata was "in control"?

Another thought: would it be better to express the extension of duration of a fermata in terms of beats rather than %age of the duration of the note to which it is attached?   My guess is that most musicians consciously or subconsciously think that way when they encounter a fermata.  So, under playback properties a fermata would have "Extend duration by [...] beats" instead of the time stretch parameter.

**Comment by wizofaus** [comment]
Yes, in that scenario, I'd expect the fermata to be based purely on the 1/4 note - i.e. the second 1/4 note beat should be stretched by 50%.  But you can always change that 150% to be more or less as needed.
In a more complex scenario where where you had a 1/2 note with fermata in one staff/voice, and in another you have a 1/4 note with a fermata and then a 1/4 note without, it would be the first beat that would be stretched by 50%.
But if you deleted all those 1/4 notes, then the 1/2 note itself would be stretched by 150% (which would result in a longer time stretch).
What's a "beat"?  What if you had the above but the time signature was 6/8?

**Comment by SteveBlower** [comment]
What's a beat -  deep question... it all depends  ... 

So yes, my suggestion is half baked.  Let me put it back in the oven for a while.

**Comment by shoogle** [comment]
I think we should somehow encourage the user to add the fermata to the stave that has the shortest note during the "pause". This becomes the **_controlling fermata_** that determines the time stretch for all staves.

Once the controlling fermata is added, fermatas would appear on other staves automatically, but the time stretch would only last for the duration of the shortest note.

If the user wants other notes to be slow too, they must add fermatas to those notes as well, and potentially make them invisible. Extra fermatas would only appear on other staves if they don't overlap with the original fermata.

**Comment by shoogle** [comment]
Here's a tricky one:

<img width="281" height="246" alt="Image" src="https://github.com/user-attachments/assets/cffa05ee-bc1d-4814-b830-bc7a1e52d9c4" />

<br>The default behaviour should be:

* Beat 1 played at normal speed.
* Beat 2 played more slowly.
* Beat 3 played at normal speed again.

This example shows:

1. Time stretch should apply **not** to the shortest note or rest, but rather the _shortest region of overlap_.
    - This region starts on the note or rest that starts latest in the score (i.e. the highest start tick).
    - It ends on the note or rest that finishes earliest in the score (i.e. the lowest end tick).
3. Regardless of which fermata the user adds first, it's _not possible_ for MuseScore to guess with 100% accuracy which note the second fermata should be added to.
    - In ambiguous cases like this, the user must be allowed to choose which notes (or rests) are to receive fermatas.
    - Only those notes will be slowed, and only for the shortest region of overlap.

If the user wants something else, they can add additional fermatas, and potentially make them invisible.

<img width="290" height="236" alt="Image" src="https://github.com/user-attachments/assets/dcee4f13-79c8-47db-84bd-f64ffc47bf76" />

**Comment by SteveBlower** [comment]
It is only tricky because it is incorrect, or at least unhelpful notation.  Fermatas indicate that the end of the note (or rest) is delayed.  They should be notated for all instruments to indicate the notes **end points**, and they should all occur at the same time.  If a user writes something ambiguous it is not Musescore's job to work out what the user means and what it does in such nonsensical cases is irrelevant.  Assuming the intention is to hold beat two (which is what is shown in the flute part) then the end of beat two has to be shown in all parts.  This is simply done using a tie from beats two to three in the oboe part with the fermata on the first quarter note.  If the intention is to hold beat three (which is what is indicated in the oboe part), then the flute player has to have a fermata on beat three not on beat two.

Whatever, this is getting away from the initial point of the issue which is that having a 200% default length for a fermata means that all fermatas have to be adjusted if the user wants a different length.  Musescore (helpfully) only honours the longest fermata. Therefore having a 100% default length - i.e. not prolonging the note, makes it easy to get the desired effect by setting the length of any single fermata at that point in the score.  Any others at the same point can be left at the non-extending default.
